packages/config/src/options.js

```diff
@@ -191,6 +191,13 @@ export function getNuxtConfig(_options) {
     options.render.csp = defaults(isObject(csp) ? csp : {}, cspDefaults)
   }
 
+  // Check if options.render is truthy
+  if (options.render.csp) {
+    if (options.render.csp.hasOwnProperty('asMeta')) {
+      // ToDo
```

Comment:
Is this the right implementation :thinking:

Reply:
I would go back to

` if (this.context.options.render.csp && this.context.options.render.csp.asMeta) {`

Reply:
> For the last "logic" part (enable on generate by default), you have to use `options.js`, yes

- What you suggested above had already been brought about within `renderer.js`
- Now, in order to enable it on generate by default I had made an initial workaround as given above :arrow_up: 
- It goes within `options.js` (it checks whether `options.render.csp` is truthy and if is the case proceed with the check for existence of the `asMeta` property within the `render.csp` object.)
- What should go within the nested if block :thinking: 

Hope you got the sense of where am I stuck at :smiley:

Reply:
@manniL 

> For the last "logic" part (enable on generate by default), you have to use `options.js`, yes

- What you suggested above had already been brought about within `renderer.js`
- Now, in order to enable it by default on `nuxt generate` I had made an initial workaround as given above :arrow_up: 
- It goes within `options.js` (checks whether `options.render.csp` is truthy and if that's the case proceed with the check for existence of the `asMeta` property within the `render.csp` object.)
- What should go within the nested if block :thinking: 

Hope you got the sense of where I'm stuck at :smiley:

Reply:
Whoops, I mixed up the files, sorry! 🙈

Reply:
What we want for `options` is:

1. Check if CSP is truthy
2. Check if in "generation mode"
3. enable `asMeta` then.

Reply:
How are we gonna check whether if it is in the `generate` mode :thinking: 
Currently I can see an `options.modes` array, but not sure if it is the required one to be updated as per context.

> 3. Check if asMeta is strictly set to false, if so, return (explicitly disabled)
> 4. enable asMeta otherwise.
Is this the required workaround :-
```js
if (options.render.csp.asMeta === false) {
  return
} else {
     // Enable asMeta 
}
```
I'm not sure on what do you mean by enabling `asMeta`
Initially the `csp` object lacks that property right?

Reply:
The only possibility to detect generate mode is `process.static` which is only filled during generate. We may use this value to force enable `asMeta` inside rendered. /cc @manniL @jamesgeorge007

Reply:
OR: Set `options._generate` from CLI/commands/generate

Reply:
> I'm not sure on what do you mean by enabling asMeta
 Initially the csp object lacks that property right?

If `render.csp` is truthy, is *should* be an object. Everything else doesn't make sense. If so, we can set the `asMeta` key easily.

Regarding your first question, @pi0 	already presented two options which you can consider.

> `if (options.render.csp.asMeta === false) {
  return`

That wasn't meant literally, sorry :see_no_evil: 	I mean that we should skip the force activation if it's set to false explicitly. Returning early would lead to way more skipping as it's a huge fn (which we may should split up :thinking: )